torch/utils/data/dataset.py

```diff
@@ -260,6 +265,133 @@ def __len__(self):
         return len(self.indices)
 
 
+class ChunkDataReader(object):
```

Comment:
Something to discuss. ChunkDataReader is implemented at dataset.py, which only has datasets... Is it the best place to host this class?

Reply:
Could we create a chunk.py and put all the classes in that file?, it could be under dataset

Reply:
Yes, that is a good choice. However, the `torch/utils/data` folder has only 4 major files: `dataset.py`, `dataloader.py`, `distributed.py` and `sampler.py`. With exception of `distributed.py`, all other files are major deep learning concepts, so Facebook might not like to put too specific files there. There is also a `_utils` folder with helper functions, which I dont recomend for public interface files - they should have more visibility.

Another idea is to move `ChunkDataReader` to `distributed.py` as - ultimately - it will be used in distributed training.

Reply:
@jaliyae @xzhu1900

Reply:
keeping at distributed.py